x-pack/plugins/timelines/public/components/t_grid/body/index.tsx

```diff
@@ -233,19 +235,63 @@ export const BodyComponent = React.memo<StatefulBodyProps>(
       [additionalControls, browserFields, columnHeaders, id]
     );
 
-    const [sortingColumns, setSortingColumns] = useState([]);
+    const sortingColumns: Array<{
+      id: string;
+      direction: 'asc' | 'desc';
+    }> = useMemo(
+      () =>
+        sort.map((x) => ({
+          id: x.columnId,
+          direction: mapSortDirectionToDirection(x.sortDirection),
+        })),
+      [sort]
+    );
 
     const onSort = useCallback(
-      (columns) => {
-        setSortingColumns(columns);
+      (
+        nextSortingColumns: Array<{
+          id: string;
+          direction: 'asc' | 'desc';
+        }>
+      ) => {
+        dispatch(
+          tGridActions.updateSort({
+            id,
+            sort: mapSortingColumns({ columns: nextSortingColumns, columnHeaders }),
+          })
+        );
+
+        setTimeout(() => {
+          // schedule the query to be re-executed from page 0, (but only after the
+          // store has been updated with the new sort):
+          if (loadPage != null) {
+            loadPage(0);
+          }
+        }, 0);
       },
-      [setSortingColumns]
+      [columnHeaders, dispatch, id, loadPage]
     );
 
     const [visibleColumns, setVisibleColumns] = useState(() =>
       columnHeaders.map(({ id: cid }) => cid)
     ); // initializes to the full set of columns
 
+    const onSetVisibleColumns = useCallback(
+      (newVisibleColumns: string[]) => {
+        const removed = columnHeaders.filter((h) => !newVisibleColumns.includes(h.id));
+
+        removed.forEach((c) =>
+          dispatch(
+            tGridActions.removeColumn({
+              id,
+              columnId: c.id,
+            })
+          )
+        );
+      },
+      [columnHeaders, dispatch, id]
+    );
```

Comment:
Testing it locally I noticed the column reordering stoped working, from both the _columns_ popover and from the header "move left" and "move right" actions.
I think the cause is we need to call `setVisibleColumns(newVisibleColumns)` inside this function.

Reply:
Great catch @semd! Removing this custom `onSetVisibleColumns` (because we're reverting to the default behavior of the grid) resolves this